app/components/UI/Stake/components/StakingBalance/StakingBalance.test.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { fireEvent, screen } from '@testing-library/react-native';
+import { renderScreen } from '../../../../../util/test/renderWithProvider';
+import { backgroundState } from '../../../../../util/test/initial-root-state';
+import StakingBalance from './StakingBalance';
+import { strings } from '../../../../../../locales/i18n';
+import Routes from '../../../../../constants/navigation/Routes';
+
+function render(Component: React.ComponentType) {
+  return renderScreen(
+    Component,
+    {
+      name: 'Asset',
+    },
+    {
+      state: {
+        engine: {
+          backgroundState,
+        },
+      },
+    },
+  );
+}
+
+const mockNavigate = jest.fn();
+
+jest.mock('@react-navigation/native', () => {
+  const actualReactNavigation = jest.requireActual('@react-navigation/native');
+  return {
+    ...actualReactNavigation,
+    useNavigation: () => ({
+      navigate: mockNavigate,
+    }),
+  };
+});
+describe('StakingBalance', () => {
+  it('should render and match snapshot', () => {
+    render(StakingBalance);
+    expect(screen.toJSON()).toMatchSnapshot();
+  });
+
+  it('should redirect to StakeInputView when the stake button is clicked', () => {
```

Comment:
Should we change the test description to match guidelines [here](https://github.com/MetaMask/contributor-docs/blob/main/docs/unit-testing.md#use-it-to-specify-the-desired-behavior-for-the-code-under-test) ?

```suggestion
  it('redirects to StakeInputView on stake button click', () => {
```